app/genesis.go

```diff
@@ -13,8 +15,25 @@ import (
 // object provided to it during init.
 type GenesisState map[string]json.RawMessage
 
+const (
+	// DefaultMaxWasmCodeSize limit max bytes read to prevent gzip bombs
+	// 600 KB is copied from x/wasm, but you can customize here as desired
+	DefaultMaxWasmCodeSize = 600 * 1024 * 2
+)
+
 // NewDefaultGenesisState generates the default state for the application.
 func NewDefaultGenesisState() GenesisState {
 	encCfg := MakeEncodingConfig()
-	return ModuleBasics.DefaultGenesis(encCfg.Marshaler)
+	gen := ModuleBasics.DefaultGenesis(encCfg.Marshaler)
+
+	// here we override wasm config to make it permissioned by default
+	wasmGen := wasm.GenesisState{
+		Params: wasmtypes.Params{
+			CodeUploadAccess:             wasmtypes.AllowNobody,
+			InstantiateDefaultPermission: wasmtypes.AccessTypeEverybody,
+			MaxWasmCodeSize:              DefaultMaxWasmCodeSize,
+		},
```

Comment:
Oh cool! So when we do the upgrade to cosmwasm, do we need to override these in InitGenesis as well?

Heres what we had to do for `x/txfees` in our v5 upgrade: https://github.com/osmosis-labs/osmosis/blob/main/app/upgrades/v5/upgrades.go#L60-L66

Reply:
Yeah, we will need a migration step, that will set the params explicitly.
I wanted to do that on another PR. I could copy your migration setup, but would be happier to add a few lines to an existing one (if you could set up the framework for v6 upgrade)